platform/macos/src/MGLMapView.mm

```diff
@@ -1618,11 +1618,18 @@ - (BOOL)gestureRecognizer:(NSGestureRecognizer *)gestureRecognizer shouldAttempt
 #pragma mark Keyboard events
 
 - (void)keyDown:(NSEvent *)event {
-    if (event.modifierFlags & NSNumericPadKeyMask) {
+    NSEventModifierFlags modFlags = event.modifierFlags;
+    BOOL noModFlags = modFlags == 256;
+    BOOL isZoomEvent = ([@[@"-", @"="] containsObject:[event characters]] && noModFlags) ||
+                        ([@"+" isEqual:[event characters]] && (modFlags & NSShiftKeyMask));
+
+    if (modFlags & NSNumericPadKeyMask) {
         // This is the recommended way to handle arrow key presses, causing
         // methods like -moveUp: and -moveToBeginningOfParagraph: to be called
         // for various standard keybindings.
         [self interpretKeyEvents:@[event]];
+    } else if (isZoomEvent) {
+        [self adjustZoomLevelForKey:[event characters]];
```

Comment:
I am not very much familiar with macOS programming (let alone keyboard event handling). I'll update the code to reflect your recommendation.